src/app/components/normal-component/lobby/lobby.component.spec.ts

```diff
@@ -25,39 +28,426 @@ describe('LobbyComponent', () => {
+            .and.returnValue(new BehaviorSubject(list).asObservable());
+    }
+    describe('clicking on a started game', () => {
 
-        // When clicking on the part
-        await testUtils.clickElement('#part_0');
+        let unstartedPartUserCreated: PartDocument;
+        let unstartedPartUserDidNotCreate: PartDocument;
+        let anotherUnstartedPartUserDidNotCreate: PartDocument;
+
+        let startedPartUserPlay: PartDocument;
+        let startedPartUserDoNotPlay: PartDocument;
+        let anotherStartedPartUserDoNotPlay: PartDocument;
+
+        beforeEach(fakeAsync(async() => {
+            // Unstarted
+            unstartedPartUserCreated = new PartDocument('I-create', PartMocks.INITIAL);
+            unstartedPartUserDidNotCreate = new PartDocument('I-did-not-create', PartMocks.OTHER_UNSTARTED);
+            anotherUnstartedPartUserDidNotCreate = new PartDocument('me-no-create-either', PartMocks.OTHER_UNSTARTED);
+
+            // Started
+            startedPartUserPlay = new PartDocument('I-play', PartMocks.STARTED);
+            startedPartUserDoNotPlay = new PartDocument('I-do-not-play', PartMocks.OTHER_STARTED);
+            anotherStartedPartUserDoNotPlay = new PartDocument('me-no-play-either', PartMocks.ANOTHER_STARTED);
+        }));
+        describe('as a user participating to no games', () => {
+            beforeEach(() => {
+                // Given an user not part of any part
+                ConnectedUserServiceMock.setObservedPart(MGPOptional.empty());
+            });
+            it('Should redirect to /play', fakeAsync(async() => {
+                // And a server with one active part
+                setLobbyPartList([startedPartUserDoNotPlay]);
+                const router: Router = TestBed.inject(Router);
+                spyOn(router, 'navigate').and.resolveTo();
+                testUtils.detectChanges();
+
+                // When clicking on the part
+                await testUtils.clickElement('#part_0');
+
+                // Then the component should navigate to the part
+                expectValidRouting(router, ['/play', 'Quarto', startedPartUserDoNotPlay.id], OnlineGameWrapperComponent);
+            }));
+        });
+        describe('as a player', () => {
+            beforeEach(() => {
+                // Given an user observing a part as a Player
+                ConnectedUserServiceMock.setObservedPart(MGPOptional.of({
+                    id: startedPartUserPlay.id,
+                    role: 'Player',
+                    typeGame: 'P4',
+                }));
+            });
+            it('should forbid user to join another game', fakeAsync(async() => {
+                // And a lobby where the a part user do not play is present
+                setLobbyPartList([startedPartUserDoNotPlay]);
+                testUtils.detectChanges();
+
+                // When clicking on the part
+                spyOn(component.messageDisplayer, 'criticalMessage').and.callThrough();
```

Comment:
en vrai pour éviter tous ces tick(3000), on pourrait faire du .and.resolveTo() ici